charts/kube-otel-stack/values.yaml

```diff
@@ -244,9 +251,190 @@ metricsCollector:
+            timestamp:
+              parse_from: time
+              layout: "%Y-%m-%dT%H:%M:%S.%LZ"
+          # Parse Docker format
+          - type: json_parser
+            id: parser-docker
+            output: extract_metadata_from_filepath
+            timestamp:
+              parse_from: time
+              layout: "%Y-%m-%dT%H:%M:%S.%LZ"
+          # Extract metadata from file path
+          - type: regex_parser
+            id: extract_metadata_from_filepath
+            regex: '^.*\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[a-f0-9\-]{36})\/(?P<container_name>[^\._]+)\/(?P<run_id>\d+)\.log$'
+            parse_from: $$attributes.file_path
+          # Move out attributes to Attributes
+          - type: metadata
+            attributes:
+              stream: "EXPR($.stream)"
+              k8s.container.name: "EXPR($.container_name)"
+              k8s.namespace.name: "EXPR($.namespace)"
+              k8s.pod.name: "EXPR($.pod_name)"
+              run_id: "EXPR($.run_id)"
+              k8s.pod.uid: "EXPR($.uid)"
+          # Clean up log record
+          - type: restructure
+            id: clean-up-log-record
+            ops:
+              - remove: logtag
+              - remove: stream
+              - remove: container_name
+              - remove: namespace
+              - remove: pod_name
+              - remove: run_id
+              - remove: uid
+    processors:
+      memory_limiter:
+        check_interval: 1s
+        limit_percentage: 75
+        spike_limit_percentage: 30
+      resourcedetection/env:
+        detectors: [env]
+        timeout: 2s
+        override: false
+      batch:
+        send_batch_size: 1000
+        timeout: 1s
+        send_batch_max_size: 1500
+      k8sattributes:
+        passthrough: false
+        pod_association:
+          - sources:
+              - from: resource_attribute
+                name: k8s.pod.name
+        extract:
+          metadata:
+            - k8s.namespace.name
+            - k8s.pod.name
+            - k8s.pod.uid
+            - k8s.node.name
+            - k8s.pod.start_time
+            - k8s.deployment.name
+            - k8s.replicaset.name
+            - k8s.replicaset.uid
+            - k8s.daemonset.name
+            - k8s.daemonset.uid
+            - k8s.job.name
+            - k8s.job.uid
+            - k8s.cronjob.name
+            - k8s.statefulset.name
+            - k8s.statefulset.uid
+            - container.image.tag
+            - container.image.name
+      resource:
+        attributes:
+          - key: lightstep.helm_chart
+            value: kube-otel-stack
+            action: insert
+          - key: collector.name
+            value: "${KUBE_POD_NAME}"
+            action: insert
+
+    exporters:
+      elasticsearch:
+        endpoints: ["https://logingest.lightstep.com"]
+        index: "lightstep"
+        headers:
+          "lightstep-access-token": "${LS_TOKEN}"
+
+    service:
+      pipelines:
```

Comment:
we should also be enabling self scrape metrics here too probably.

Reply:
Can you elaborate more on this? None of the other collectors scrape themselves, so I'm not sure what I should change.

Reply:
For the metric collectors, the there's a receiver that's automatically added for scraping collector meta-monitoring metrics here: https://github.com/lightstep/otel-collector-charts/blob/main/charts/kube-otel-stack/templates/collector.yaml#L87-L93

If we add a metrics pipeline w/ the receiver code snippet above to the new log collector, will be able to get good monitoring metrics for it.

Reply:
Added with 6e1ee4d8214a6fb5271b0378d035dd4c2689d13c

Reply:
We should probably come up with a better solution here as well, as I could see someone accidentally removing the `prometheus` pipelines thinking they were added by accident (since there is no `prometheus` receiver defined in the values file). Either specifying all pipelines in the values file or pushing the whole self-scrape pipeline to `collector.yaml`.